crates/task-impls/src/quorum_proposal/handlers.rs

```diff
@@ -166,22 +167,28 @@ impl<TYPES: NodeType, V: Versions> ProposalDependencyHandle<TYPES, V> {
 
         let version = self.upgrade_lock.version(self.view_number).await?;
 
-        let block_header = if version < V::Marketplace::VERSION {
+        let high_qc = self.consensus.read().await.high_qc().clone();
+
+        let builder_commitment = commitment_and_metadata.builder_commitment.clone();
+        let metadata = commitment_and_metadata.metadata.clone();
+        let get_legacy_block_header_fn = async {
             TYPES::BlockHeader::new_legacy(
                 state.as_ref(),
                 self.instance_state.as_ref(),
                 &parent_leaf,
                 commitment_and_metadata.commitment,
-                commitment_and_metadata.builder_commitment,
-                commitment_and_metadata.metadata,
+                builder_commitment,
+                metadata,
                 commitment_and_metadata.fees.first().clone(),
                 vid_share.data.common.clone(),
                 version,
             )
             .await
             .wrap()
-            .context(warn!("Failed to construct legacy block header"))?
-        } else {
+            .context(warn!("Failed to construct legacy block header"))
+        };
+
+        let get_marketplace_block_header_fn = async {
```

Comment:
nit: I think this could just be called `marketplace_block_header` (similarly `legacy_block_header`)

(I think the `_fn` confused me slightly because it's just a block header future)

Reply:
I feel like it's nice to have `_fn` here so that we know we'll call `.await` to really get the value. No strong opinion though!